src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -350,31 +404,38 @@ void FullyConnected::setDynamicBatchLim(int lim) {
         primArgs.at(argType) = dnnl::memory(newMemDesc, oldMem.get_engine(), oldMem.get_data_handle());
     };
 
-    setBatchPrimArgs(DNNL_ARG_SRC, getParentEdgesAtPort(0)[0]->getMemory().GetPrimitive());
-    setBatchPrimArgs(DNNL_ARG_DST, getChildEdgesAtPort(0)[0]->getMemory().GetPrimitive());
+    if (useConv1x1) {
+        Node::setDynamicBatchLim(lim);
+    } else {
+        dynBatchLim = lim;
+        setBatchPrimArgs(DNNL_ARG_SRC, getParentEdgesAtPort(0)[0]->getMemory().GetPrimitive());
+        setBatchPrimArgs(DNNL_ARG_DST, getChildEdgesAtPort(0)[0]->getMemory().GetPrimitive());
+    }
 }
 
 void FullyConnected::execute(dnnl::stream strm) {
-    if (prim) {
-        // in cases parameter -> FullyConnected or dynamic shapes
-        // we keep old pointer to data in primArgs on second iteration with same input shapes
-        auto updateMemoryPtr = [this](int argType) {
-            auto param = primArgs.find(argType);
-            if (param != primArgs.end()) {
-                if (argType == DNNL_ARG_SRC && getInputShapeAtPort(DATA_ID).getRank() == 3) {
-                    primArgs.at(argType).set_data_handle(getParentEdgesAtPort(0)[0]->getMemoryPtr()->GetData());
-                }
-                if (argType == DNNL_ARG_DST && getOutputShapeAtPort(0).getRank() == 3) {
-                    primArgs.at(argType).set_data_handle(getChildEdgesAtPort(0)[0]->getMemoryPtr()->GetData());
-                }
+    if (!execPtr) {
+        IE_THROW() << "Can't execute FullyConnected node with name: " << getName() << ", because executor is not compiled";
+    }
+
+    // in cases parameter -> FullyConnected or dynamic shapes
+    // we keep old pointer to data in primArgs on second iteration with same input shapes
+    auto updateMemoryPtr = [this](int argType) {
+        auto param = primArgs.find(argType);
+        if (param != primArgs.end()) {
+            if (argType == DNNL_ARG_SRC && (getInputShapeAtPort(DATA_ID).getRank() == 3 || useConv1x1)) {
```

Comment:
If we switched from Conv1x1 to IP on this iteration `useConv1x1` will be false as I can see. Shouldn't we updated memory anyway?

Reply:
If rank==3 the memory will update here when `useConv1x1` changed to false, it should be ok. If rank==2 the memory will be updated to parent output `dnnl::memory` object and share the same memory with parent in `prepareParams`, it should also be ok:
https://github.com/openvinotoolkit/openvino/blob/738d7bb09fedd17398e0ff70a591c27bc1e14e67/src/plugins/intel_cpu/src/nodes/fullyconnected.cpp#L342-L345

I will add test to cover this case.

Reply:
Testcase added in #14144 .